src/openttd.cpp

```diff
@@ -1078,7 +1078,22 @@ void SwitchToMode(SwitchMode new_mode)
 			MakeNewEditorWorld();
 			break;
 
-		case SM_RESTARTGAME: // Restart --> 'Random game' with current settings
+		case SM_RESTARTGAME: // Restart --> Current settings preserved
+			if (_file_to_saveload.abstract_ftype == FT_SAVEGAME || _file_to_saveload.abstract_ftype == FT_SCENARIO) {
```

Comment:
>I made some changes to only intercept savegames in editor mode. Would that be satisfactory?

I don't think it would, tbh. Would just be a confusing edgecase.

Besides, when could this actually happen? You can't load savegames into the editor, only scenarios (even if they're just savegames renamed to .scn extension)

Reply:
You refer to a reply from 13 months ago, on a codebase which has changed since then.
I do not understand your comment in the context on the discussion you quote, applied to unrelated code?
What are you expecting to see?